### PR TITLE
refactor fusion util for sparse ops

### PR DESCRIPTION
**Pull request by Orion34-lanbo**
1. Separate where fusion logic from `fusion_utils.cc`
2. set up `SparseFusionStrategy` for where fusion and upcoming `kSparseReduction` fusion

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/fusion_utils.h [hunk 1]
```diff
@@ -863,6 +868,20 @@ class TransformBasedCpuFusionStrategy : public FusionStrategy {
   }
 };
 
+class SparseOpCpuFusionStrategy : public FusionStrategy {
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/fusion_utils.h:871]: nit: better to add a filecheck based ut?

**Review by qiuxiafei — COMMENTED** [review]
LGTM, but pls add some file-check ?

File: tao_compiler/mlir/disc/transforms/fusion_utils.h (hunk 1 above)
Comment (Orion34-lanbo) [reply tao_compiler/mlir/disc/transforms/fusion_utils.h:871]: done

**Review by qiuxiafei — APPROVED** [review]

**Comment by Orion34-lanbo** [comment]
to #510

**Pull request merged**